internal/netxlite/utls.go

```diff
@@ -46,22 +46,22 @@ type utlsConn struct {
 	nc net.Conn
 }
 
-// Ensures that a utlsConn implements the TLSConn interface.
-var _ TLSConn = &utlsConn{}
+// Ensures that a UTLSConn implements the TLSConn interface.
+var _ TLSConn = &UTLSConn{}
 
-// newConnUTLS returns a NewConn function for creating utlsConn instances.
-func newConnUTLS(clientHello *utls.ClientHelloID) func(conn net.Conn, config *tls.Config) (TLSConn, error) {
+// NewConnUTLS returns a NewConn function for creating UTLSConn instances.
+func NewConnUTLS(clientHello *utls.ClientHelloID) func(conn net.Conn, config *tls.Config) (TLSConn, error) {
```

Comment:
The name of this function was ~fine as long as it was private. Now that it is public, I am wondering why its name seems to imply that it creates a new connection, while instead it _returns_ a function for creating a new connection. I suppose we can probably find a better name. Additionally, or instead of doing that, we could also consider keeping this function private. It seems to me an adapter to allow creating uTLS connections inside `netxlite/tls.go` and I think most users want instead to call `NewConnUTLSWithHelloID`. Regardless of what we choose to do, perhaps `NewConnUTLS` is the right name for `NewConnUTLSWithHelloID`?

Reply:
> It seems to me an adapter to allow creating uTLS connections inside netxlite/tls.go and I think most users want instead to call NewConnUTLSWithHelloID.

I have the same understanding.

> perhaps NewConnUTLS is the right name for NewConnUTLSWithHelloID?

Makes sense. 
And I agree, the adapter should rather be private. I can't think of a better name for it though, but I'll think about it

Reply:
Maybe 
`NewConnUTLSWithHelloID` --> `NewUTLSConn`,
`NewConnUTLS` --> `newConnWithUTLS`

Reply:
How about:

`NewConnUTLSWithHelloID` --> `NewUTLSConn`,
`newConnUTLS` --> `newUTLSConnFactory`

Not exactly as before but at least it removes the ambiguity of two function only differing in their first letter capitalization and doing a different business?

Reply:
Yes, I think that's good 👍